### PR TITLE
[MLv2] Drop non-standard normalization in QB selectors

### PR DESCRIPTION
**Pull request by bshepherdson**
Use the canonical `metabase.mbql.js/normalize` instead.

Currently this contains both versions and compares results; I'll
remove that after a clean CI run.

DO NOT SUBMIT

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/29830)

### CODE REVIEW

**Review by camsaul — APPROVED** [review]
> Reviewed 8 of 8 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/metabase/metabase/29830)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @kulyk)

File: frontend/src/metabase-lib/util.ts
Comment (kulyk) [inline frontend/src/metabase-lib/util.ts:1]: Could we rename the file to `utils` (plural), please? 🙏 I think we prefer the plural form on the FE

File: frontend/src/metabase-lib/util.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import * as ML from "cljs/metabase.lib.js";
+import type { DatasetQuery } from "metabase-types/api";
+
+export function areQueriesEqual(
+  query1: DatasetQuery,
+  query2: DatasetQuery,
+  fieldIds?: number[],
+): boolean {
+  return ML.query_EQ_(query1, query2, fieldIds);
+}
```
Comment (kulyk) [inline frontend/src/metabase-lib/util.ts:10]: Do you think we'd ever need to compare opaque pMBQL queries? If so, would it make sense to reserve `areQueriesEqual` name for them and use something like `areLegacyQueriesEqual` in this case?

File: frontend/src/metabase/query_builder/selectors.unit.spec.js [hunk 2]
```diff
@@ -133,6 +142,41 @@ describe("getIsResultDirty", () => {
       expect(getIsResultDirty(state)).toBe(true);
     });
 
+    it("converts clauses into plain MBQL objects", () => {
+      const aggregation = ["count"];
+      const breakout = ["field", ORDERS.CREATED_AT.id, null];
+      const filter = [">=", ["field", ORDERS.TOTAL.id, null], 20];
```
Comment (kulyk) [inline frontend/src/metabase/query_builder/selectors.unit.spec.js:148]: fwiw, `metabase-lib/metadata/Field` instances should have a `reference` method that could give you a simple field ref. It should be possible to replace this with something like that:

```jsx
const breakout = ORDERS.CREATED_AT.reference()
const filter = [">=", ORDERS.TOTAL.reference(), 20]
```

File: frontend/src/metabase/query_builder/selectors.unit.spec.js [hunk 3]
```diff
@@ -133,6 +142,41 @@ describe("getIsResultDirty", () => {
...
+      const orderBy = ["asc", ["aggregation", 0]];
+      const join = {
+        alias: "Products",
+        condition: [
+          "=",
+          ["field", ORDERS.PRODUCT_ID.id, null][
+            ("field", PRODUCTS.ID.id, null)
+          ],
+        ],
+      };
```
Comment (kulyk) [inline frontend/src/metabase/query_builder/selectors.unit.spec.js:158]: Could you please clarify what does the condition mean here? 

```js
["field", ORDERS.PRODUCT_ID.id, null][
  ("field", PRODUCTS.ID.id, null)
]
```

I'm not sure I've seen anything like that in MBQL before

**Review by bshepherdson — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/metabase/metabase/29830)* status: all files reviewed, 4 unresolved discussions (waiting on @kulyk)

___
*[`frontend/src/metabase/query_builder/selectors.unit.spec.js` line 148 at r1](https://reviewable.io/reviews/metabase/metabase/29830#gh-1161647746:-NSfG0po0I9afPP_MUli:b5a83d0) ([raw file](https://github.com/metabase/metabase/blob/8ab812fa70aa16b21e6dc4a2f9cdd0875224e15e/frontend/src/metabase/query_builder/selectors.unit.spec.js#L148)):*
<details><summary><i>Previously, kulyk (Anton Kulyk) wrote…</i></summary><blockquote>

fwiw, `metabase-lib/metadata/Field` instances should have a `reference` method that could give you a simple field ref. It should be possible to replace this with something like that:

```jsx
const breakout = ORDERS.CREATED_AT.reference()
const filter = [">=", ORDERS.TOTAL.reference(), 20]
```
</blockquote></details>

Oh, handy. Fixed.
___
*[`frontend/src/metabase/query_builder/selectors.unit.spec.js` line 158 at r1](https://reviewable.io/reviews/metabase/metabase/29830#gh-1161648679:-NSfxCY5A_WlD-bmvKSH:b-hrl9l8) ([raw file](https://github.com/metabase/metabase/blob/8ab812fa70aa16b21e6dc4a2f9cdd0875224e15e/frontend/src/metabase/query_builder/selectors.unit.spec.js#L158)):*
<details><summary><i>Previously, kulyk (Anton Kulyk) wrote…</i></summary><blockquote>

Could you please clarify what does the condition mean here? 

```js
["field", ORDERS.PRODUCT_ID.id, null][
  ("field", PRODUCTS.ID.id, null)
]
```

I'm not sure I've seen anything like that in MBQL before
</blockquote></details>

Done. This was a bad edit. I'm actually surprised it parsed and ran this test successfully.
___
*[`frontend/src/metabase-lib/util.ts` line 0 at r1](https://reviewable.io/reviews/metabase/metabase/29830#gh-1161646163:-NSgpsRI3uJ_93kMNHuW:b-896fix) ([raw file](https://github.com/metabase/metabase/blob/8ab812fa70aa16b21e6dc4a2f9cdd0875224e15e/frontend/src/metabase-lib/util.ts#L0)):*
<details><summary><i>Previously, kulyk (Anton Kulyk) wrote…</i></summary><blockquote>

Could we rename the file to `utils` (plural), please? 🙏 I think we prefer the plural form on the FE
</blockquote></details>

Done.
___
*[`frontend/src/metabase-lib/util.ts` line 10 at r1](https://reviewable.io/reviews/metabase/metabase/29830#gh-1161646764:-NSgq-eT6RFQf2QRAU-5:b-yh16l6) ([raw file](https://github.com/metabase/metabase/blob/8ab812fa70aa16b21e6dc4a2f9cdd0875224e15e/frontend/src/metabase-lib/util.ts#L10)):*
<details><summary><i>Previously, kulyk (Anton Kulyk) wrote…</i></summary><blockquote>

Do you think we'd ever need to compare opaque pMBQL queries? If so, would it make sense to reserve `areQueriesEqual` name for them and use something like `areLegacyQueriesEqual` in this case?
</blockquote></details>

Done. Good call.

**Review by camsaul — APPROVED** [review]
> Reviewed 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/metabase/metabase/29830)* status: all files reviewed, 4 unresolved discussions (waiting on @kulyk)

**Review by kulyk — APPROVED** [review]
LGTM 🚢

**Pull request merged**